src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.CompilationManager.cs

```diff
@@ -24,7 +24,7 @@ internal partial class DiagnosticAnalyzerService
     /// value if the set of analyzers changes.  In that case, a new instance will be created and will be cached for the
     /// next caller.
     /// </summary>
-    private static readonly ConditionalWeakTable<Project, StrongBox<(ImmutableArray<DiagnosticAnalyzer> analyzers, CompilationWithAnalyzersPair? compilationWithAnalyzersPair)>> s_projectToCompilationWithAnalyzers = new();
+    private static readonly ConditionalWeakTable<ProjectState, StrongBox<(ImmutableArray<DiagnosticAnalyzer> analyzers, CompilationWithAnalyzersPair? compilationWithAnalyzersPair)>> s_projectToCompilationWithAnalyzers = new();
```

Comment:
>StrongBox [](http://example.com/codeflow?start=63&length=9)

Had this comment in the earlier PR, but it had already merged. I'm curious why a strongbox is used here rather than just a standard Tuple.

Reply:
It's a CWT. Value, unfortunately, cannot be a struct.  It's an API limitation.

Reply:
Tuple, not ValueTuple

Reply:
Oh. I just don't care though you use that type. Esp as you can't name args. :-)

Reply:
Ahh, the naming ability does provide value there. Makes sense, thanks!